### DISCUSSION TITLE
How to Manage Roles and Permissions between two different tables within same package while using Laravel Nova Multi Tenant Package

### DISCUSSION THREAD

**Discussion by BilalMajeedDarvis** [pktharindu/nova-permissions#63] [Q&A] [answered]
In multi tenant there is a central model known as "admin"  and then tenant model known as "user" . The migrations are separate for tenant and central. 
My question is how can we manage this package for admin ? I have change the table thru nova-permission.php as well and then ran the migration and it run perfectly but when i created the role in the admin it's giving me the following error. 
"Illuminate\ Database \ QueryException
SQLSTATE[42S02]: Base table or view not found: 1146 Table 'estatfolio.users' doesn't exist"

How can i redirect it to "estatfolio.admins" ?? 
Please Help.

**Comment by pktharindu** [comment, marked as answer]
You'll have to change the `user_model` in the config as well as the `table_names`.

**Reply by BilalMajeedDarvis** [reply]
MY tabke_names
'table_names' => [
        'roles' => 'roles',

        'role_permission' => 'role_permission',

        'role_user' => 'role_user',

        'users' => 'users',
        'admins' => 'admins',
    ],
    
Like this 
'user_model' => 'App\Models\User',
    'user_model' => 'App\Models\Admin',
    
    OR THIS
    
    'user_model' => 'App\Models\User',
    'admin_model' => 'App\Models\Admin',

**Reply by pktharindu** [reply]
Yeah, that's not going to work. This package only supports one user model at a time. You might be able to extend this to suit your needs. Although I'm not sure why you would need two tables/models for that. The admin can simply be another role you attach to a user.

Which multi-tenancy package are you using?

**Reply by BilalMajeedDarvis** [reply]
I am using this packages https://tenancyforlaravel.com/saas-boilerplate/
It does uses two tables. Can you make any work around that?

**Reply by pktharindu** [reply]
Yeah, I think it is possible to extend this package to accommodate that. But unfortunately, I have very limited availability and don't have a licence for that boilerplate to even try it out 😅

**Reply by BilalMajeedDarvis** [reply]
Can you suggest me any work way around it please? I need to implement this.

**Reply by pktharindu** [reply]
I don't think there is a way around it. Basically, you'll have to adjust the config to accept multiple `user_models`, `user_resources` and `users` table and maybe make the `role_user` table polymorphic. Then you'll have to adjust pretty much everything else facilitate the changes.
